northstar/src/runtime/state.rs

```diff
@@ -841,120 +686,27 @@ impl<'a> State<'a> {
             .collect()
     }
 
-    /// Mount all the containers in every repository
-    async fn mount_repositories(&mut self) -> Result<(), Error> {
-        let mut mounted_containers = Vec::new();
-
-        // Mount all the containers from each repository
-        for repo in self.repositories.keys() {
-            let mut containers = self.mount_repository(&repo).await?;
-            mounted_containers.append(&mut containers);
-        }
-
-        for container in mounted_containers {
-            self.add(container)?;
-        }
-
-        Ok(())
+    async fn notification(&self, n: Notification) {
+        self.events_tx
+            .send(Event::Notification(n))
+            .await
+            .expect("Internal channel error on main");
     }
+}
 
-    /// Mounts all the `npk` files in the specified repository
-    async fn mount_repository(&self, repo_id: &str) -> Result<Vec<Container>, Error> {
-        let repo = self.repositories.get(repo_id).ok_or_else(|| {
-            Error::RepositoryIdUnknown(
-                repo_id.to_string(),
-                self.repositories().keys().map(|id| id.into()).collect(),
-            )
-        })?;
-
-        info!("Mounting NPKs from {}", repo.dir.display());
-        let mut dir = fs::read_dir(&repo.dir).await.map_err(|e| {
-            Error::Io(
-                format!(
-                    "Failed to read repository at {}",
-                    &repo.dir.to_string_lossy()
-                ),
-                e,
-            )
-        })?;
-        let mut containers = Vec::new();
-
-        while let Ok(Some(entry)) = dir.next_entry().await {
-            let container = self
-                .mount_control
-                .mount_npk(&entry.path(), repo, &self.config.run_dir)
-                .await
-                .map_err(Error::Mount)?;
-            containers.push(container);
-        }
+/// Dump the hello world npk created at compile time into a tmpdir that acts as internal
+/// repository.
+#[cfg(debug_assertions)]
+async fn prepare_internal_repository(name: &str) -> Result<(tempfile::TempDir, Repository), Error> {
```

Comment:
internal repository is not a good description...rather demo repository or s.th. like that

Reply:
It's in `debug` builds only so it's not important: `demo` or `example`?

Reply:
Think we can do this in an follow up.

Reply:
Created #307